### DISCUSSION TITLE
Replacing / adding attributes / buffers after running vsgconv on an .ive

### DISCUSSION THREAD

**Discussion by dchristopherfennell** [vsg-dev/VulkanSceneGraph#1124] [Q&A] [answered]
Hello,

What is the proper way to do something like this?

I use vsgconv to convert an .ive to .vsgb.

The .vsgb model will now have everything I need, but I need to swap out the shaders and add a light. I can swap out the shaders using 
a visitor, but I'm stuck binding to a global directional light (lightData). I can see my light data being set in ViewDependentState.cpp, 
but the values in my shaders are 0.

I want to have the light UBO accessable to all shaders that bind to the UBO. 

I've tried different methods up to this point, based on some of the examples (vsglights, vsgclip, etc) and from what I have read in the discussions, but I can never access the light data from my shader.

Thanks,
DC

**Comment by robertosfield** [comment, marked as answer]
Right now osg2vsg doesn't generated shaders set up that utilizes the modern VSG features like vsg::ViewDependentState and the various ShaderSet that support lighting, shadows etc.

What is required is a rewrite of osg2vsg to use the modern VSG features so that when OSG files are converted to VSG they are already in a form that leverages all the modern features.
